File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String str : tokens) {
            
            if (str.equals("+")) {
                int num2 = stack.pop();
                int num1 = stack.pop();
                stack.add(num1+num2);
            } else if (str.equals("-")) {
                int num2 = stack.pop();
                int num1 = stack.pop();
                stack.add(num1-num2);
            } else if (str.equals("*")) {
                int num2 = stack.pop();
                int num1 = stack.pop();
                stack.add(num1*num2);
            } else if (str.equals("/")) {
                int num2 = stack.pop();
                int num1 = stack.pop();
                stack.add(num1/num2);
            } else {
                stack.add(Integer.parseInt(str));
            }
        }
        return stack.pop();
    }
}

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token == "+":
                stack.append(stack.pop() + stack.pop());
            elif token == "-":
                stack.append(-stack.pop() + stack.pop());
            elif token == "*":
                stack.append(stack.pop() * stack.pop());
            elif token == "/":
                nb2, nb1 = stack.pop(), stack.pop()
                stack.append(int(nb1/nb2));
            else:
                stack.append(int(token))
        
        return stack.pop()